core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -719,7 +741,8 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
   private def confirmExportedAction(actionLookup: Future[WhiskActionMetaData], authenticated: Boolean)(
     implicit transid: TransactionId): Future[WhiskActionMetaData] = {
     actionLookup flatMap { action =>
-      val requiresAuthenticatedUser = action.annotations.getAs[Boolean]("require-whisk-auth").exists(identity)
+      val requiresAuthenticatedUser =
+        action.annotations.getAs[Boolean](WhiskAction.requireWhiskAuthAnnotation).exists(identity)
```

Comment:
I was thinking... not sure if it works... to make the auth check a truthy, so that then you can consolidate the authentication failure. The `authenticated` bit could be computed by the earlier check of the header (or the whisk key as done now).

Reply:
let's discuss after the refactoring into a separate method.. @markusthoemmes wanted to be sure we minimized the deserialization of the json values.